pkg/operator/vsphere_check.go

```diff
@@ -304,9 +305,23 @@ func newClient(ctx context.Context, cfg *vsphere.VSphereConfig, username, passwo
 	tctx, cancel := context.WithTimeout(ctx, *check.Timeout)
 	defer cancel()
 	klog.V(4).Infof("Connecting to %s as %s, insecure %t", serverAddress, username, insecure)
+
+	// Set user to nil there for prevent login during client creation.
+	// See https://github.com/vmware/govmomi/blob/master/client.go#L91
+	serverURL.User = nil
```

Comment:
Can you remove or replace line 302, instead of overwriting it?
https://github.com/openshift/vsphere-problem-detector/blob/dccdaec77e1369e611a4c08a7a0bb0107013217a/pkg/operator/vsphere_check.go#L302